pkg/controller/bundle_sync_task.go

```diff
@@ -199,106 +281,128 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
 		return false, processErr
 	}
 
-	// Construct resource conditions and check if there were any resource errors
-	resourceStatuses := make([]smith_v1.ResourceStatus, 0, len(st.processedResources))
-	var failedResources []smith_v1.ResourceName
-	retriableResourceErr := true
-	statusUpdated := false
-	for _, res := range st.bundle.Spec.Resources { // Deterministic iteration order
-		blockedCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceBlocked, Status: smith_v1.ConditionFalse}
-		inProgressCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceInProgress, Status: smith_v1.ConditionFalse}
-		readyCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceReady, Status: smith_v1.ConditionFalse}
-		errorCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceError, Status: smith_v1.ConditionFalse}
-
-		if resInfo, ok := st.processedResources[res.Name]; ok {
-			// Resource was processed
-			switch resStatus := resInfo.status.(type) {
-			case resourceStatusDependenciesNotReady:
-				blockedCond.Status = smith_v1.ConditionTrue
-				blockedCond.Reason = smith_v1.ResourceReasonDependenciesNotReady
-				blockedCond.Message = fmt.Sprintf("Not ready: %q", resStatus.dependencies)
-			case resourceStatusInProgress:
-				inProgressCond.Status = smith_v1.ConditionTrue
-			case resourceStatusReady:
-				readyCond.Status = smith_v1.ConditionTrue
-			case resourceStatusError:
-				errorCond.Status = smith_v1.ConditionTrue
-				errorCond.Message = resStatus.err.Error()
-				if resStatus.isRetriableError {
-					errorCond.Reason = smith_v1.ResourceReasonRetriableError
+	bundleUpdated := false
+
+	if st.newFinalizers != nil {
+		// Update finalizers
+		st.bundle.Finalizers = st.newFinalizers
+		// Set status to "InProgress: True"
+		// (there will be one more iteration for resource sync)
+		// TODO: add more details to the status (reason, resources status)?
+		st.bundle.Status = smith_v1.BundleStatus{
+			Conditions: []smith_v1.BundleCondition{
+				{Type: smith_v1.BundleInProgress, Status: smith_v1.ConditionTrue},
```

Comment:
Here is where we'd change the status to something like delete in progress. @ash2k